backend/src/controllers/filter-item-controller.ts

```diff
@@ -0,0 +1,101 @@
+import { Request, Response } from 'express'; 
+import { validationResult } from 'express-validator'; 
+import InventoryItem from '../models/inventory-item'; 
+import Item from '../models/item';
+import mongoose from 'mongoose';
+
+interface ItemInterface {
+    _id: string,
+    name: string,
+    description: string,
+    pickupLocation: string,
+    pickupTime: string
+}
+
+interface FilterInterface {
+    _id?: any;
+    itemOwner?: string; 
+    name?: any
+}
+
+export const getFilteredInventoryItems = async (req: Request, res: Response) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        console.log("errors", errors);
+        return res.status(422).json({ errors: errors.array() });
+    }
+
+    try {
+        // Filter
+        let fetchedInventoryItems: any = {};
+        const itemOwner = req.query.itemOwner as string
```

Comment:
`itemOwner` filter isn't working. I'll release a patch in the prep work prior to the App Showcase